packages/components/src/mobile/audio-player/test/audio-url-parser.native.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Internal dependencies
+ */
+import { parseAudioUrl } from '../audio-url-parser.native';
+
+const supportedAudioUrlsWithExtensions = [
+	'https://www.mp3.com/file.mp3?key1=value1&key2=value2#anchorforsomereason',
+	'https://www.mp3.com/file.mp3?key1=value1&key2=value2',
+	'https://www.mp3.com/file.mp3',
+];
+
+const supportedAudioUrlsWithoutExtensions = [
+	'https://www.mp3.com/file?key1=value1&key2=value2#anchorforsomereason',
+	'https://www.mp3.com/file?key1=value1&key2=value2',
+	'https://www.mp3.com/file',
+	'https://www.mp3.com/folder/file',
+];
+
+describe( 'supportedAudioUrlsWithExtensions', () => {
+	supportedAudioUrlsWithExtensions.forEach( ( url ) => {
+		it( `supports ${ url }`, () => {
+			const { title, extension } = parseAudioUrl( url );
+			expect( title ).not.toBe( '' );
+			expect( extension ).not.toBe( '' );
+		} );
+	} );
+} );
+
+describe( 'supportedAudioUrlsWithoutExtensions', () => {
+	supportedAudioUrlsWithoutExtensions.forEach( ( url ) => {
+		it( `supports ${ url }`, () => {
+			const { title, extension } = parseAudioUrl( url );
+			expect( title ).not.toBe( '' );
+			expect( extension ).toBe( '' );
```

Comment:
Similar to my previous comment, we could check here specific values, for example:

URL: `https://www.mp3.com/folder/file`
```
expect( title ).toBe( 'file' );
expect( extension ).toBe( '' );
```

Reply:
Included here e4ea54b